Sources/SourceKitLSP/SourceKitServer.swift

```diff
@@ -1298,13 +1313,126 @@ extension SourceKitServer {
         return index.occurrences(ofUSR: usr, roles: roles)
       }
 
-      req.reply(extractedResult)
+      req.reply(extractedResult.map { $0.map(\.location) })
+    }
+    let request = Request(symbolInfo, id: req.id, clientID: ObjectIdentifier(self),
+                          cancellation: req.cancellationToken, reply: callback)
+    languageService.symbolInfo(request)
+  }
+
+  private func indexToLSPCallHierarchyItem(symbol: Symbol, location: Location) -> CallHierarchyItem {
+    CallHierarchyItem(
+      name: symbol.name,
+      kind: symbol.kind.asLspSymbolKind(),
+      tags: nil,
+      uri: location.uri,
+      range: location.range,
+      selectionRange: location.range,
+      // We encode usr and uri for incoming/outgoing call lookups in the implementation-specific data field
+      data: .dictionary([
+        "usr": .string(symbol.usr),
+        "uri": .string(location.uri.stringValue),
+      ])
```

Comment:
We encode `usr` and `uri` in the (implementation-specific) LSP data field since incoming/outgoing call requests are only provided with a `CallHierarchyItem` in the request parameters.